cms/djangoapps/contentstore/views/course.py

```diff
@@ -519,11 +551,15 @@ def format_library_for_view(library):
             u'can_edit': has_studio_write_access(request.user, library.location.library_key),
         }
 
-    courses_iter = _remove_in_process_courses(courses_iter, in_process_course_actions)
+    courses_iter = _remove_in_process_courses(courses, in_process_course_actions)
     in_process_course_actions = [format_in_process_course_view(uca) for uca in in_process_course_actions]
+    archived_courses_iter = ()
+    if archived_courses:
+        archived_courses_iter = _remove_in_process_courses(archived_courses, in_process_course_actions)
```

Comment:
@smarnach  Yep, that does make the diff much smaller, and only adds one more iteration over the courses list when the feature is enabled.  I was trying to avoid this, but I agree that code clarity is more important here.

Fixed with 9260f01.